umh-core/pkg/diagnosis/latch_spec6_test.go

```
@@ -0,0 +1,52 @@
+// ever joins Update, this spec has been broken.
+var _ = Describe("Latch signature", func() {
+	It("should derive its state from the reduction and the window's extent, and from no readability fact of any kind", func() {
+		// Coverage is exactly two time.Duration fields (span, spanned) and
+		// nothing else, no bool, no Reading. The clear arm and re-fire arm are
+		// gated on these; a window that has already engaged its freeze (a second
+		// consecutive failed read) still spans its full duration, which is why
+		// the fields carry no fact about whether THIS tick's read succeeded.
+		ct := reflect.TypeOf(Coverage{})
+		Expect(ct.NumField()).To(Equal(2),
+			"Coverage must carry exactly two durations — a readability field smuggled in here is F1 rebuilt")
+		for i := 0; i < ct.NumField(); i++ {
```

golangci-lint (2.12.2)

[error] for loop can be changed to use an integer range (Go 1.22+)
Because the key is returned by a function or method, take care to consider side effects.

(intrange)